### PR TITLE
feat(utility-thread): SyncProvingThread now explicitly notifies UtilityThread with the current epoch building the cache up [fixes VM-594]

### PR DESCRIPTION
**Pull request by raftedproc**
This changes the way SyncThreads notifies UtilityThread about the current epoch used. This is a preparation part for the project to make CCP more interactive.

### CODE REVIEW

File: ccp/src/hashrate/handler.rs [hunk 1]
```diff
@@ -87,4 +82,16 @@ impl HashrateHandler {
...
         self.sliding_collector.hashrate()
     }
+
+    pub(crate) fn handle_epoch(&self, epoch: EpochParameters) -> HResult<()> {
+        let mut guard = self.collector.lock().unwrap();
+        if let EpochObservation::EpochChanged {
+            prev_epoch_hashrate,
+        } = guard.observe_epoch(epoch)
+        {
+            self.saver.save_hashrate_previous(prev_epoch_hashrate)?;
```
Comment (monoid) [inline ccp/src/hashrate/handler.rs:92]: The guard is still hold here.  Is it really needed?

Comment (raftedproc) [reply ccp/src/hashrate/handler.rs:92]: You are saving the current hashrate + cleaning the things up. Doesn't this worth a lock?

Comment (monoid) [reply ccp/src/hashrate/handler.rs:92]: Then it should be lock on `saver`.

Comment (raftedproc) [reply ccp/src/hashrate/handler.rs:92]: After some discussion in slack I am stating the lock granularity here should be addressed by VM-589.

**Review by monoid — APPROVED** [review]
utACK

**Pull request merged**